pkg/controller/backup_configuration.go

```diff
@@ -0,0 +1,356 @@
+			}
+		}
+	case workload_api.KindReplicationController:
+		if resource, err := c.rcLister.ReplicationControllers(backupconfiguration.Namespace).Get(resource_name); err == nil {
+			key, err := cache.MetaNamespaceKeyFunc(resource)
+			if err == nil {
+				c.rcQueue.GetQueue().Add(key)
+			}
+		}
+	case workload_api.KindReplicaSet:
+		if resource, err := c.rsLister.ReplicaSets(backupconfiguration.Namespace).Get(resource_name); err == nil {
+			key, err := cache.MetaNamespaceKeyFunc(resource)
+			if err == nil {
+				c.rsQueue.GetQueue().Add(key)
+			}
+		}
+
+	}
+
+}
+
+func (c *StashController) EnsureCronJob(backupconfiguration *api_v1beta1.BackupConfiguration) error {
+	image := docker.Docker{
+		Registry: c.DockerRegistry,
+		Image:    docker.ImageStash,
+		Tag:      c.StashImageTag,
+	}
+
+	meta := metav1.ObjectMeta{
+		Name:      backupconfiguration.Name,
+		Namespace: backupconfiguration.Namespace,
+	}
+
+	if c.EnableRBAC {
+		ref, err := reference.GetReference(scheme.Scheme, backupconfiguration)
+		if err != nil {
+			return err
+		}
+		if err := c.ensureCronJobRBAC(ref); err != nil {
+			return fmt.Errorf("error ensuring rbac for kubectl cron job %s, reason: %s", meta.Name, err)
+		}
+	}
+
+	_, _, err := batch_util.CreateOrPatchCronJob(c.kubeClient, meta, func(in *v1beta1.CronJob) *v1beta1.CronJob {
+		// set backup-configuration as cron-job owner
+		in.OwnerReferences = []metav1.OwnerReference{
+			{
+				APIVersion: api_v1beta1.SchemeGroupVersion.String(),
+				Kind:       api_v1beta1.ResourceKindBackupConfiguration,
+				Name:       backupconfiguration.Name,
+				UID:        backupconfiguration.UID,
+			},
+		}
+		in.Spec.Schedule = backupconfiguration.Spec.Schedule
+		if in.Spec.JobTemplate.Labels == nil {
+			in.Spec.JobTemplate.Labels = map[string]string{}
+		}
+		in.Spec.JobTemplate.Labels["app"] = util.AppLabelStash
+		in.Spec.JobTemplate.Spec.Template.Spec.Containers = core_util.UpsertContainer(
+			in.Spec.JobTemplate.Spec.Template.Spec.Containers,
+			core.Container{
+				Name:            backupconfiguration.Name,
+				ImagePullPolicy: "Always",
+				Image:           image.ToContainerImage(),
+				Args: []string{
+					"backup_session",
+					fmt.Sprintf("--backupSessionName=%s", backupconfiguration.Name),
+					fmt.Sprintf("--backupSessionNamespace=%s", backupconfiguration.Namespace),
+				},
+			})
+		in.Spec.JobTemplate.Spec.Template.Spec.RestartPolicy = core.RestartPolicyNever
+		in.Spec.JobTemplate.Spec.Template.Spec.ServiceAccountName = in.Name
+		return in
+
+	})
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (c *StashController) EnsureCronJobDeleted(namespace, name string) error {
+	backupconfiguration, err := c.bupcLister.BackupConfigurations(namespace).Get(name)
+	ref, err := reference.GetReference(scheme.Scheme, backupconfiguration)
+	if err != nil {
+		return err
+	}
+	meta := metav1.ObjectMeta{
+		Name:      name,
+		Namespace: namespace,
+		OwnerReferences: []metav1.OwnerReference{
```

Comment:
Why setting `OwnerReferences:` here? Won't `core_util.EnsureOwnerReference(&meta, ref)` will it?

Reply:
fixed